Use 1-thread pool executor for subsession event handlers

qyryq opened the pull request.

Commit:
Use 1-thread pool executor for subsession event handlers

alexnick88 commented on ydb/public/sdk/cpp/client/ydb_federated_topic/impl/federated_write_session.cpp:
```diff
@@ -122,22 +124,20 @@ void TFederatedWriteSessionImpl::OpenSubsessionImpl(std::shared_ptr<TDbInfo> db)
 
     auto handlers = NTopic::TWriteSessionSettings::TEventHandlers()
-        .HandlersExecutor(Settings.EventHandlers_.HandlersExecutor_)
+        .HandlersExecutor(NTopic::CreateThreadPoolExecutor(1))
```
This must be one executor per driver. What will happen if client creates 1000 federated write sessions? This code will create 1000 threads.

alexnick88 commented on ydb/public/sdk/cpp/client/ydb_federated_topic/impl/federated_write_session.cpp:
```diff
@@ -122,22 +124,20 @@ void TFederatedWriteSessionImpl::OpenSubsessionImpl(std::shared_ptr<TDbInfo> db)
-                TDeferredWrite deferred;
-
                 with_lock(self->Lock) {
+                    TDeferredWrite deferred;
```
TDeferredWrite is not functional anymore. Remove this class.

Commit:
Federated write session: use the same subsession handlers executor for each TDbDriverStatePtr

Commit:
Delete TDeferredWrite

As we use our own single-threaded event handlers executor,
deadlocks should not happen there, so we don't need TDeferredWrite.

Commit:
One SubsessionHandlersExecutor per TFederatedTopicClient::TImpl
